### PR TITLE
Feat(eos_cli_config_gen): Add support for OSPF graceful restart

### LINKED ISSUE #4468: Feat(eos_cli_config_gen): Add support for OSPF graceful restart
### Enhancement summary

add dedicated keys under `router_ospf` for manipulating graceful restart

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Configuring the graceful restart feature for OSPF.

### Describe the solution you would like

I would like to suggest dedicated keys under router_ospf like:

router_ospf:
  graceful_restart : <bool>
  graceful_restart_grace_period: <int:1-1800> # seconds
  graceful_restart_helper: <bool> # Default: True
  
  This should render
  
```
router ospf 100
   graceful-restart
   
```

or 
```
router ospf 100
   graceful-restart grace-period 15
   no graceful-restart-helper
```

### Describe alternatives you have considered

This can be implemented today using eos_cli:

```
custom_structured_configuration_router_ospf:
        process_ids:
          100:
            eos_cli: graceful-restart
```

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Add support for OSPF graceful restart

## Related Issue(s)

Fixes #4468 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
schema - 

```
router_ospf:
     process_ids:
         - graceful_restart:
                 enabled: <bool>

                 # Specify maximum time in seconds to wait for graceful-restart to complete.
                 grace_period: <int; 1-1800>
           graceful_restart_helper: <bool>
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-ospf.j2 [hunk 1]
```diff
@@ -200,6 +200,16 @@ router ospf {{ process_id.id }}
    summary-address {{ summary_address.prefix }}
 {%         endif %}
 {%     endfor %}
+{%     if process_id.graceful_restart.enabled is arista.avd.defined(true) %}
+   graceful-restart
+{%     elif process_id.graceful_restart.grace_period is arista.avd.defined %}
+   graceful-restart grace-period {{ process_id.graceful_restart.grace_period }}
+{%     endif %}
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-ospf.j2]:
```suggestion
{%     if process_id.graceful_restart.enabled is arista.avd.defined(true) %}
{%         set graceful_cmd = "graceful-restart" %}
{%         if process_id.graceful_restart.grace_period is arista.avd.defined %}
{%.            graceful_cmd = graceful_cmd ~ " grace-period " ~ process_id.graceful_restart.grace_period %}
{%.        endif %}
   {{ graceful_cmd }}
{%     endif %}
```
so enabled will be required key

Comment (Shivani-gslab) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-ospf.j2]: Fixed

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-ospf.j2 [hunk 2]
```diff
@@ -200,6 +200,18 @@ router ospf {{ process_id.id }}
...
 {%     endfor %}
+{%     if process_id.graceful_restart.enabled is arista.avd.defined(true) %}
+{%         set graceful_restart_cli = "graceful-restart" %}
+{%         if process_id.graceful_restart.grace_period is arista.avd.defined %}
+{%             set graceful_restart_cli = graceful_restart_cli ~ " grace-period " ~ process_id.graceful_restart.grace_period %}
+{%         endif %}
+   {{ graceful_restart_cli }}
+{%     endif %}
+{%     if process_id.graceful_restart_helper is arista.avd.defined(true) %}
+   graceful-restart-helper
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-ospf.j2:214]: wan2-dc1(config-router-ospf)#sh ac
router ospf 11
   max-lsa 12000
   summary-address 10.1.1.1/32
   graceful-restart grace-period 10
   mpls ldp sync default
   no graceful-restart-helper
   
   seems config is not in correct order

Comment (Shivani-gslab) [reply python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-ospf.j2:214]: Fixed

**Review by MaheshGSLAB — APPROVED** [review]

**Review by laxmikantchintakindi — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]
Blocking merge until 5.0 has been released.

**Review by ClausHolbechArista — DISMISSED** [review]
Blocking merge until 5.0 has been released.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**